pkg/sensors/tracing/kprobe_test.go

```diff
@@ -5095,3 +5097,123 @@ spec:
 		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
 	}
 }
+
+// Test module loading/unloading on Ubuntu
+func TestTraceKernelModule(t *testing.T) {
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	hookFull := `apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "monitor-kernel-modules"
+spec:
+  kprobes:
+  - call: "security_kernel_read_file"
+    # Explicit module loading using file descriptor finit_module() to print module full path
+    syscall: false
+    return: true
+    args:
+    - index: 0
+      type: "file"
+    - index: 1
+      type: "int"
+    returnArg:
+      index: 0
+      type: "int"
+    selectors:
+    - matchArgs:
+      - index: 1
+        operator: "Equal"
+        values:
+        - "2"  # READING_MODULE
+  - call: "do_init_module"
+    syscall: false
+    args:
+    - index: 0
+      type: "module"
+  - call: "free_module"
+    syscall: false
+    args:
+    - index: 0
+      type: "module"
+`
+
+	// This test works only on Ubuntu
+	f, err := os.Open("/etc/os-release")
+	if err != nil {
+		t.Skip("Skipping test: could not parse /etc/os-release file")
+	}
+	defer f.Close()
+
+	ubuntu := false
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), "Ubuntu") {
+			ubuntu = true
+			break
+		}
+	}
+
+	if ubuntu == false {
+		t.Skip("Skipping test: could not determin if this is an Ubuntu machine")
+	}
+
+	createCrdFile(t, hookFull)
+
+	obs, err := observertesthelper.GetDefaultObserverWithFile(t, ctx, testConfigFile, tus.Conf().TetragonLib)
+	if err != nil {
+		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
+	}
+	observertesthelper.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+	readyWG.Wait()
+
+	module := "xfs"
```

Comment:
this looks generic to me, what's ubuntu specific?

Reply:
@olsajiri I checked quickly our test vms that we generate and xfs is not compiled in. There are some networking modules but setting these up as quick examples may need to install extra packages... so I think this is good enough as first coverage for tests.

I added it here as a follow up: https://github.com/cilium/tetragon/issues/1433